### PR TITLE
Use const_cast instead of c style cast

### PR DESCRIPTION
**Pull request by arnavb**

### CODE REVIEW

File: src/openrct2-cli/Cli.cpp [hunk 1]
```diff
@@ -27,7 +27,7 @@ using namespace OpenRCT2;
 int main(int argc, char * * argv)
```
Comment (janisozaur) [inline src/openrct2-cli/Cli.cpp]: Can't the `argv` here be marked as `const`?

Comment (arnavb) [reply src/openrct2-cli/Cli.cpp]: @janisozaur I don't think you can change the signature of main.

Comment (janisozaur) [reply src/openrct2-cli/Cli.cpp]: I'm pretty sure you can, see https://github.com/OpenRCT2/OpenRCT2/pull/6493

Comment (arnavb) [reply src/openrct2-cli/Cli.cpp]: Ok, if you say so, it just seems unusual to change the signature that is widely used. I would personally keep the cast.

Comment (janisozaur) [reply src/openrct2-cli/Cli.cpp]: It is widely used, but it is not specified by the standard. The `argv` data is known to be coming from OS and should not really be touched, in fact, we had an issue due to that way in the past. We have yet to see if all the CIs come green on that other PR.

Comment (arnavb) [reply src/openrct2-cli/Cli.cpp]: Should I close this then?

Comment (janisozaur) [reply src/openrct2-cli/Cli.cpp]: I'm happy for you to take credit, you can update this PR to match changes done in #6493 or I can mention you in the commit message. Up to you.

Comment (arnavb) [reply src/openrct2-cli/Cli.cpp]: Ok, I will update here.

**Comment by arnavb** [comment]
@janisozaur Ok, I am done now.

**Comment by janisozaur** [comment]
@IntelOrca can you also do the review of Windows part?

**Comment by arnavb** [comment]
@janisozaur What do you mean by "needs squashing"?

**Comment by janisozaur** [comment]
@arnavb when merging, we will squash it.

**Pull request merged**